wiki/development/coding-guidelines.md

```diff
@@ -80,127 +238,233 @@ class Two {foo = 2;};
 
-There also exists the FUNC family of Macros:
+### 5.6. Magic Numbers
+There should be no magic numbers. Any magic number should be put in a define either on top of the .sqf file (below the header), or in the script_component.hpp file in the root directory of the component (recommended) in case it is used in multiple locations.
 
-| Macros  |  Expands to |
-| -------|---------|
-|`FUNC(face)` | ace_balls_fnc_face or the call trace wrapper for that function.|
-|`EFUNC(balls,face)` | ace_balls_fnc_face or the call trace wrapper for that function.|
-|`EFUNC(leg,face) `| ace_leg_fnc_face or the call trace wrapper for that function.|
-|`DFUNC(face)` | ace_balls_fnc_face and will ALWAYS be the function global variable.|
-|`DEFUNC(leg,face)` | ace_leg_fnc_face and will ALWAYS be the function global variable.|
-|`QFUNC(face)` | "ace_balls_fnc_face" |
-|`QEFUNC(leg,face)` | "ace_leg_fnc_face" |
+Magic numbers are any of the following:
+* A constant numerical or text value used to identify a file format or protocol
+* Distinctive unique values that are unlikely to be mistaken for other meanings
+* Unique values with unexplained meaning or multiple occurrences which could (preferably) be replaced with named constants
 
-The `FUNC` and `EFUNC` macros should NOT be used inside `QUOTE` macros if the intention is to get the function name or assumed to be the function variable due to call tracing (see below). If you need to 100% always be sure that you are getting the function name or variable use the `DFUNC` or `DEFUNC` macros. For example `QUOTE(FUNC(face)) == "ace_balls_fnc_face"` would be an illegal use of `FUNC` inside `QUOTE`.
+[Source](http://en.wikipedia.org/wiki/Magic_number_%28programming%29)
 
-Using `FUNC` or `EFUNC` inside a `QUOTE` macro is fine if the intention is for it to be executed as a function.
+## 6. Code Standards
 
-#### 4.1.1 FUNC Macros, Call Tracing, and Non-ACE3 /Anonymous Functions
+### 6.1. Error testing
+If a function returns error information, then that error information will be tested.
 
-ACE3 implements a basic call tracing system that can dump the call stack on errors or wherever you want. To do this the `FUNC` macros in debug mode will expand out to include metadata about the call including line numbers and files. This functionality is automatic with the use of calls via `FUNC` and `EFUNC`, but any calls to other functions need to use the following macros:
+### 6.2. Unreachable Code
+There shall be no unreachable code.
 
-| Macro  | example |
-| -------|---------|
-|`CALLSTACK(functionName)` | `[] call CALLSTACK(cba_fnc_someFunction)` |
-|`CALLSTACK_NAMED(function,functionName)` | `[] call CALLSTACK_NAMED(_anonymousFunction,'My anonymous function!')`|
+### 6.3. Function Parameters
+Parameters of functions must be retreived through the usage of the param or params commands. If the function is part of the public API, parameters must be checked on allowed data types and values through the usage of the param and params commands.
```

Comment:
Code tags around `param` and `params`, though I'd just say `params`, `param` is not useful as much as I heard.